### PR TITLE
Nj 34 add county code

### PR DESCRIPTION
**Pull request by aloverso**
## Link to issue
- https://github.com/newjersey/affordability-pm/issues/34

## Is PM acceptance required? (delete one)
- Yes - don't merge until issue is accepted!

## What was done?
- Add CSV for NJ municipalities/counties/codes and parsing logic
- Add County dropdown question
- Add Municipality dropdown question based on County answer
- Add NJ Review step with ability to edit municipality/county
- Add MunicipalityCode to NJ XML under "CountyCode"
  
## Screen recording

https://github.com/user-attachments/assets/3ce73634-1c7c-40c4-8743-7043a742aae6

### CODE REVIEW

File: app/forms/state_file/nj_county_form.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  class NjCountyForm < QuestionsForm
+    set_attributes_for :intake,
+                       :county
+
+    validates :county, presence: true
+
+    def save
+      if @intake.county != self.county
+        @intake.municipality_code = nil
+        @intake.municipality_name = nil
```
Comment (aloverso) [inline app/forms/state_file/nj_county_form.rb:11]: I'd like to write a test for this behavior (when the county is changed, the municipality code/name should get reset) but I'm not sure how to do so in the context of how the form specs are structured. How could I do that?

File: spec/models/efile_error_spec.rb [hunk 2]
```diff
@@ -60,6 +60,9 @@
       "federal-info",
       "initiate-data-transfer",
       "name-dob",
+      "nj-county",
```
Comment (mmazanec22) [inline spec/models/efile_error_spec.rb:63]: Hadn't realized we needed to add this! Added a note to our internal doc

File: app/controllers/state_file/questions/nj_county_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+require 'csv'
```
Comment (mmazanec22) [inline app/controllers/state_file/questions/nj_county_controller.rb:1]: [Dust] Can the require statement live in NjMunicipalities instead?

File: app/lib/efile/nj/nj_municipalities.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          load
+          @municipalities_by_county.keys.sort
+        end
+
+        def find_name_by_county_and_code(county, code)
+          load
+          county_rows = @municipalities_by_county[county]
+          raise KeyError.new("Invalid county") unless county_rows
+          municipality = county_rows.find { |muni| muni.code == code }
+          raise KeyError.new("Invalid code") unless municipality
```
Comment (mpidcock) [inline app/lib/efile/nj/nj_municipalities.rb:15]: [DUST] I might include "municipality" in this message, as "Invalid municipality code" or "Invalid municipality" as a good practice, but it's incredibly minor here

**Review by mpidcock — APPROVED** [review]
Looks great! Nice work

File: app/views/state_file/questions/nj_county/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    content_for :page_title, title
+%>
+
+<% content_for :card do %>
+  <h1 class="h2"><%= title %></h1>
+
+
+  <%= form_with model: @form, url: { action: :update }, local: true, method: "put", builder: VitaMinFormBuilder do |f| %>
+    <div class="white-group">
+      <%= f.vita_min_state_file_select(:county , t(".county"), Efile::Nj::NjMunicipalities.county_options, label_class: 'form-label--bold', prompt: t('general.select_prompt') ) %>
```
Comment (mpidcock) [inline app/views/state_file/questions/nj_county/edit.html.erb:12]:
```suggestion
      <%= f.vita_min_state_file_select(:county, t(".county"), Efile::Nj::NjMunicipalities.county_options, label_class: 'form-label--bold', prompt: t('general.select_prompt') ) %>
```

File: app/views/state_file/questions/nj_municipality/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<% title = t(".title", filing_year: Rails.configuration.statefile_current_tax_year) %>
+<% content_for :page_title, title %>
+
+<% content_for :card do %>
+  <h1 class="h2"><%= title %></h1>
+
+  <%= form_with model: @form, url: { action: :update }, local: true, method: "put", builder: VitaMinFormBuilder do |f| %>
+    <div class="white-group">
+      <%= f.vita_min_state_file_select(
+            :municipality_code ,
```
Comment (mpidcock) [inline app/views/state_file/questions/nj_municipality/edit.html.erb:10]:
```suggestion
            :municipality_code,
```

**Review by mpidcock — COMMENTED** [review]
I spotted some stray spaces, could be nice to fix up

**Comment by mrotondo** [comment]
FYI @aloverso I just merged https://github.com/codeforamerica/vita-min/pull/4735 which makes some changes to the `return_to_review` shared example, so you should be able to delete the explicit nj case from that file - we just missed it during the StateInformationService refactor.

File: spec/factories/state_file_nj_intakes.rb [hunk 7]
```diff
@@ -29,6 +30,8 @@
 #  locale                            :string           default("en")
 #  locked_at                         :datetime
 #  message_tracker                   :jsonb
+#  municipality_code                 :string
+#  municipality_name                 :string
```
Comment (mmazanec22) [inline spec/factories/state_file_nj_intakes.rb:34]: [dust] Why save the code and the name?

**Review by mmazanec22 — APPROVED** [review]

**Review by jachan — APPROVED** [review]
Thanks for the great example of how to add a question! One small non-blocking comment about how county code is formatted.

**Pull request merged**